File: app/src/main/java/ardjomand/leonardo/arcmovies/data/MoviesRepositoryImpl.java
package ardjomand.leonardo.arcmovies.data;

import ardjomand.leonardo.arcmovies.data.remote.MoviesRemoteRepository;

/**
 * Created by unity on 25/02/18.
 */

public class MoviesRepositoryImpl implements MoviesRepository {

    private final MoviesRepository mRemoteRepository = new MoviesRemoteRepository();

    @Override
    public void loadUpcomingMovies(LoadUpcomingMoviesCallback loadUpcomingMoviesCallback, int page) {
        mRemoteRepository.loadUpcomingMovies(loadUpcomingMoviesCallback, page);
    }

    @Override
    public void loadMovieDetails(LoadMovieDetailsCallback loadMovieDetailsCallback, int movieId) {
        mRemoteRepository.loadMovieDetails(loadMovieDetailsCallback, movieId);
    }

    @Override
    public void loadGenres(LoadGenresCallBack loadGenresCallBack) {
        mRemoteRepository.loadGenres(loadGenresCallBack);
    }
}

File: app/src/main/java/ardjomand/leonardo/arcmovies/data/MoviesRepositoryImpl.kt
package ardjomand.leonardo.arcmovies.data

import ardjomand.leonardo.arcmovies.data.remote.MoviesRemoteRepository

/**
 * Created by unity on 25/02/18.
 */

class MoviesRepositoryImpl : MoviesRepository {

    private val mRemoteRepository = MoviesRemoteRepository()

    override fun loadUpcomingMovies(loadUpcomingMoviesCallback: MoviesRepository.LoadUpcomingMoviesCallback, page: Int) {
        mRemoteRepository.loadUpcomingMovies(loadUpcomingMoviesCallback, page)
    }

    override fun loadMovieDetails(loadMovieDetailsCallback: MoviesRepository.LoadMovieDetailsCallback, movieId: Int) {
        mRemoteRepository.loadMovieDetails(loadMovieDetailsCallback, movieId)
    }

    override fun loadGenres(loadGenresCallBack: MoviesRepository.LoadGenresCallBack) {
        mRemoteRepository.loadGenres(loadGenresCallBack)
    }
}
